tests/lexer/tstrnumlits.nim

```diff
@@ -0,0 +1,151 @@
+# Test tkStrNumLit
+
+import std/macros
+import strutils
+import mlexerutils
+
+type
+  S = distinct string
+
+proc `'wrap`(number: string): S =
+  result = ("[[" & number & "]]").S
+
+assertAST dedent """
+  StmtList
+    ProcDef
+      AccQuoted
+        Ident "\'wrap"
+      Empty
+      Empty
+      FormalParams
+        Ident "S"
+        IdentDefs
+          Ident "number"
+          Ident "string"
+          Empty
+      Empty
+      Empty
+      StmtList
+        Asgn
+          Ident "result"
+          DotExpr
+            Par
+              Infix
+                Ident "&"
+                Infix
+                  Ident "&"
+                  StrLit "[["
+                  Ident "number"
+                StrLit "]]"
+            Ident "S"""":
+  proc `'wrap`(number: string): S =
+    result = ("[[" & number & "]]").S
```

Comment:
```suggestion
    result = number
```
(simplifies the AST; the complexity of previous version is irrelevant to what we're testing in this file)

Reply:
I've removed the `S` unique type refs and simplified things.